File: Examples/InApp/IterateScenarios/Iterate scenarios in WindFarmer.py
	# Example layout iteration script
	def Execute(self):
		resultString = "Results: \n"

		for layoutScenario in Workbook.LayoutScenarios:
			Toolbox.ActivateLayoutScenario(layoutScenario)
			
			# Define some calculation settings
			energySettings = Workbook.ModelSettings.EnergySettings
			energySettings.WakeModelType = WakeModelType.EddyViscosity
			energySettings.ApplyLargeWindFarmCorrection = True
			energySettings.CalculationToUse = EnergyCalculationToUseType.New
			energySettings.CalculateEfficiencies = False
			energySettings.LargeWindFarmCorrectionSettings.BaseRoughness = 0.0002
			energySettings.LargeWindFarmCorrectionSettings.IncreasedRoughness = 0.0192
			energySettings.LargeWindFarmCorrectionSettings.DistanceInDiametersToStartOfRecovery = 120
			energySettings.NumberOfDirectionSectors = 180

			# Compute energy
			Toolbox.Log("Computing energy for layout scenario " + layoutScenario.Name, LogLevel.Warn)

			Toolbox.CalculateEnergy()

			Toolbox.Log("Finished energy calculation!", LogLevel.Warn)

			# Get results
			resultsScenario = Workbook.CurrentScenario
			subjectWindFarm = [ w for w in resultsScenario.WindFarms if w.IsNeighbour == False][0]
			Toolbox.Log("subject wind farm is: " + subjectWindFarm.Name, LogLevel.Warn)

			fullYield = resultsScenario.FarmTotalYields.GetVariantResult("Full").GetValueForFarm(subjectWindFarm).Value / 10e6	
			result = str.format("Net yield for scenario {0}:\t{1:.2f} GWh/annum\n", layoutScenario.Name, fullYield.ToString())
			Toolbox.Log(result, LogLevel.Warn)
			resultString += result

		Toolbox.Log(resultString)
		Toolbox.Log("END OF SCRIPT\n", LogLevel.Warn)
		

File: Examples/InApp/IterateScenarios/Iterate scenarios in WindFarmer.cs
/// <Summary>
/// Example layout iteration script
/// </Summary>
public void Execute()
{
	string resultString	= "Results: \n";
	
    foreach (LayoutScenario layoutScenario in Workbook.LayoutScenarios)
	{
		#region // Define some calculation settings
		var energySettings = Workbook.ModelSettings.EnergySettings;
		energySettings.WakeModelType = WakeModelType.EddyViscosity;
		energySettings.ApplyLargeWindFarmCorrection = true;
		energySettings.CalculationToUse = EnergyCalculationToUseType.New;
		energySettings.CalculateEfficiencies = false;
		energySettings.LargeWindFarmCorrectionSettings.BaseRoughness = 0.0002;
		energySettings.LargeWindFarmCorrectionSettings.IncreasedRoughness = 0.0192;
		energySettings.LargeWindFarmCorrectionSettings.DistanceInDiametersToStartOfRecovery = 120;
		energySettings.NumberOfDirectionSectors = 180;
		#endregion 
		
		// Compute energy
		Toolbox.Log("Computing energy for layout scenario " + layoutScenario.Name, LogLevel.Warn);
		Toolbox.ActivateLayoutScenario(layoutScenario);
		
		Toolbox.CalculateEnergy();
		
		Toolbox.Log("Finished energy calculation!", LogLevel.Warn);
		
		// Get results
		Scenario resultsScenario = Workbook.CurrentScenario;
		var subjectWindFarm = resultsScenario.WindFarms.Where(w => w.IsNeighbour == false).FirstOrDefault();
		Toolbox.Log("subject wind farm is: " + subjectWindFarm.Name, LogLevel.Warn);
		
		double netYield = resultsScenario.FarmTotalYields.GetVariantResult("Full").GetValueForFarm(subjectWindFarm).Value / 10e6;	
		string result = string.Format("Net yield for scenario {0}:\t{1} GWh/annum\n", layoutScenario.Name, netYield.ToString("0.##"));
		Toolbox.Log(result, LogLevel.Warn);
		resultString += result;
	}
	Toolbox.Log(resultString);
	Toolbox.Log("END OF SCRIPT", LogLevel.Warn);
}